pylint/checkers/utils.py

```diff
@@ -1778,3 +1778,17 @@ def in_for_else_branch(parent: nodes.NodeNG, stmt: nodes.Statement) -> bool:
     return isinstance(parent, nodes.For) and any(
         else_stmt.parent_of(stmt) or else_stmt == stmt for else_stmt in parent.orelse
     )
+
+
+def find_assigned_names_recursive(
```

Comment:
It feels a bit unpythonic to update in place, but that might be me. Looking at the other use-case with a small refactor using `+=` we could just make this return a list. But I'll leave that choice to you.

Reply:
We could even let this `yield` so that we exit early in `_deleted_iteration_target_cond`